app/js/components/names.jsx

```diff
@@ -1,265 +1,241 @@
+      this.setState({ errorFamilyName: 'Family Name can not be null' })
+    }
+  }
+
+  handleCancel() {
+    this.fetch();
+    const given = this.state.response.preferredName.givenName;
+    const middle = this.state.response.preferredName.middleName;
+    const family = this.state.response.preferredName.familyName;
+    this.setState((prevState) => ({
+      display: 'view',
+      givenName: given === null ? '' : prevState.givenName,
+      middleName: middle === null ? '' : prevState.middleName,
+      familyName: family === null ? '' : prevState.familyName,
+      voided: prevState.response.preferredName.voided,
+      preferred: prevState.preferred,
+      creator: prevState.response.auditInfo.creator.display,
+      dateCreated: prevState.response.auditInfo.dateCreated,
+      errorGivenName: '',
+      errorFamilyName: ''
+    }));
+  };
+
+  render() {
+    let givenErrorClass = '';
+    let familyErrorClass = '';
+    if(this.state.errorGivenName && this.state.display === 'edit') {
+      givenErrorClass = 'has-error'
+    }
+    if(this.state.errorFamilyName && this.state.display === 'edit') {
+      familyErrorClass = 'has-error'
+    }
+    return(
+      <div>
+        <form className='form-horizontal'>
+          {this.state.display === 'view' &&
+          <div className='form-group'>
+              <div className='col-sm-2'>
+                  <input className='btn btn-success form-control'
+                         name='edit'
+                         type='button'
+                         value='Edit Name'
+                         onClick={this.handleEdit}/>
+              </div>
+          </div>}
+
+          <div className='form-group '>
+              <label className='control-label col-sm-2'>Preferred</label>
+              <div className='col-sm-2'>
+                  <input
+                      name='username'
+                      type='checkbox'
+                      checked={this.state.preferred}
+                      disabled={this.state.display === 'view' ?
+                       'disabled' : null}/>
+              </div>
+          </div>
+
+          <div className={'form-group ' + givenErrorClass}>
+              <label className='control-label col-sm-2'>
+              Given Name* </label>
+              <div className='col-sm-5'>
+                  <input  className='form-control'
+                          ref='givenname'
+                          name='givenName'
+                          type='text'
+                          value={this.state.givenName}
+                          onChange={this.handleChange}
+                          readOnly={this.state.display === 'view' ?
+                           'readonly' : null}
+                          required
+                           />
+              </div>
+              {this.state.errorGivenName &&
+              this.state.display === 'edit' &&
+              <div className='input'>{this.state.errorGivenName}</div>}
```

Comment:
You may want to fix your indentation. Something like this:
`{
    this.state.errorGivenName &&
    this.state.display === 'edit' &&
    <div className='input'>
        {this.state.errorGivenName}
    </div>
}`

Reply:
Fixed

Reply:
Noted, and applied from this moment on